rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/KubernetesJsonSupport.scala

```diff
@@ -86,7 +91,41 @@ trait KubernetesJsonSupport extends SprayJsonSupport with DefaultJsonProtocol {
   implicit val podMetadataFormat: JsonFormat[PodMetadata] = jsonFormat1(PodMetadata.apply)
   implicit val podFormat: RootJsonFormat[Pod] = jsonFormat1(Pod.apply)
 
-  implicit val replicaAnnotationFormat: JsonFormat[ReplicaAnnotation] = jsonFormat1(ReplicaAnnotation.apply)
+  implicit val replicaAnnotationFormat: RootJsonFormat[ReplicaAnnotation] =
+    new RootJsonFormat[ReplicaAnnotation] {
+      // Not sure if we ever write this out, but if we do, this will let us write out exactly what we took in
+      def write(ra: ReplicaAnnotation): JsValue =
+        if (ra.revision.nonEmpty && ra.source.nonEmpty) {
+          JsObject(ra.otherAnnotations + (ra.source -> JsString(ra.revision)))
+        } else JsObject(ra.otherAnnotations)
+
+      def read(json: JsValue): ReplicaAnnotation = {
+        json match {
+          case JsObject(fields) =>
+            _revisionAnnotations.revisionAnnotations.find { annotation =>
+              fields.get(annotation).exists(_.isInstanceOf[JsString])
+            } match {
+              case Some(winningAnnotation) =>
+                ReplicaAnnotation(
+                  fields(winningAnnotation).asInstanceOf[JsString].value,
+                  winningAnnotation,
+                  fields - winningAnnotation)
+
+              case None =>
+                ReplicaAnnotation("", "", fields)
+            }
+
+          case _ => spray.json.deserializationError("expected an object")
+        }
+      }
+    }
+
   implicit val replicaSetMedatataFormat: JsonFormat[ReplicaSetMetadata] = jsonFormat1(ReplicaSetMetadata.apply)
   implicit val podReplicaSetFormat: RootJsonFormat[ReplicaSet] = jsonFormat1(ReplicaSet.apply)
 }
+
+private[kubernetes] object KubernetesJsonSupport {
+  val defaultRevisionAnnotations: HasRevisionAnnotations = new HasRevisionAnnotations {
+    val revisionAnnotations = Seq("deployment.kubernetes.io/revision")
+  }
```

Comment:
No need to hardcode the default if we put it in reference.conf as a default?

Reply:
That's basically there so there's no hard dependency (e.g. for tests) on having Config

Reply:
Hardcoding a value that is duplicated across config and code seems objectively worse than depending on config in test to me.